warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/shard/ShardedDataTypeHandler.java

```diff
@@ -679,21 +680,26 @@ protected Value createBloomFilter(RawRecordContainer event, Multimap<String,Norm
         
         if (null != maskedFieldHelper && maskedFieldHelper.contains(column)) {
             // These Keys are for the index, so if they are masked, we really want to use the normalized masked values
-            final String normalizedMaskedValue = helper.getNormalizedMaskedValue(column);
+            // It was observed that the normalized mask values aren't coming back reversed, so account for that before creating the row.
+            String normalizedMaskedValue = helper.getNormalizedMaskedValue(column);
             
             Text colf = new Text(column);
             Text colq = new Text(shardId);
             TextUtil.textAppend(colq, event.getDataType().outputName(), helper.getReplaceMalformedUTF8());
             
-            // Dont create index entries for empty values
+            // if this method was called with the intention to create reverse index keys, ensure the masked values are reversed.
             if (!StringUtils.isEmpty(normalizedMaskedValue)) {
-                // Create a key for the masked field value with the masked visibility
+                if (isReverse) {
+                    normalizedMaskedValue = new StringBuilder().append(normalizedMaskedValue).reverse().toString();
```

Comment:
Can be simplified a little: `new StringBuilder(normalizedMaskedValue).reverse().toString()`

Reply:
Alright, will adjust in the next commit.